### PR TITLE
In ScopedBuffer add reserve() and make _currentLength private and add…

### PR DESCRIPTION
**Pull request by nordlow**
… unittest.

### CODE REVIEW

File: source/mir/appender.d [hunk 1]
```diff
@@ -38,6 +38,48 @@ struct ScopedBuffer(T, size_t bytes = 4096)
         return *cast(inout(T[_bufferLength])*)&_scopeBufferPayload;
     }
 
+    ///
+    void reserve(size_t n) @trusted scope
```
Comment (9il) [inline source/mir/appender.d:42]: Can we just call `prepare` inside?

File: source/mir/appender.d [hunk 2]
```diff
@@ -26,7 +26,7 @@ struct ScopedBuffer(T, size_t bytes = 4096)
 
     private enum size_t _bufferLength =  bytes / T.sizeof + (bytes % T.sizeof != 0);
     private T[] _buffer;
-    size_t _currentLength;
+    private size_t _currentLength;
```
Comment (9il) [inline source/mir/appender.d:29]: It has to be public because it is used in some very low level weird code in mir-ion

Comment (nordlow) [reply source/mir/appender.d]: Ok. Resolved via b486406.

File: source/mir/appender.d (hunk 1 above)
Comment (nordlow) [reply source/mir/appender.d:42]: We could but that's gonna waste cycles.

Comment (nordlow) [reply source/mir/appender.d]: Resolved via d13e80f.

Comment (nordlow) [reply source/mir/appender.d]: Commented them aswell at https://github.com/libmir/mir-algorithm/compare/14347b9e0a2493c32f7fad96e538b99248cb2df3..a0902dee6bffedbe27ad0b22c75e27818e6b5774.

**Pull request merged**